plugins/callback/foreman.py

```diff
@@ -281,17 +334,40 @@ def send_reports(self, stats):
                 report['config_report']['status']['pending'] = total['changed']
                 report['config_report']['status']['applied'] = 0
 
-            self._send_data('report', host, report)
-
+            self._send_data('report', 'foreman', host, report)
             self.items[host] = []
 
+    def send_reports(self, stats):
+        if self.report_type == "foreman":
+            self.send_reports_foreman_legacy(stats)
+        elif self.report_type == "proxy":
+            self.send_reports_proxy_host_report(stats)
+        else:
+            self._display.warning(u'Unknown foreman endpoint type: {type}'.format(type=self.report_type))
+
+    def drop_nones(self, d):
```

Comment:
Is that the same as `_recursive_dict_without_none` in `plugins/module_utils/foreman_helper.py`?

https://github.com/theforeman/foreman-ansible-modules/blob/db9b4a5e61e0f0c54c8e994f91fee0586aee3bd5/plugins/module_utils/foreman_helper.py#L1675-L1693

Reply:
this comment is still open ;)

Reply:
Oh I just changed the method to also drop empty arrays/dicts. But my implementation is not perfect - it does not detect empty dict in array (`[{}]`) but I think it is still useble.

Should I move it here?

Reply:
So I can perhaps move the method to foreman_helper and import it from the callback, do you want me to do this? I am not sure how useful this will be. Maybe if you won't like dropping empty dicts/arrays then I could reuse the existing implementation.